manifests/server.pp

```diff
@@ -65,24 +66,18 @@
   anchor { 'mysql::server::end': }
 
   if $restart {
-    Anchor['mysql::server::start'] ->
-    Class['mysql::server::install'] ->
-    # Only difference between the blocks is that we use ~> to restart if
-    # restart is set to true.
     Class['mysql::server::config'] ~>
-    Class['mysql::server::service'] ->
-    Class['mysql::server::root_password'] ->
-    Class['mysql::server::providers'] ->
-    Anchor['mysql::server::end']
-  } else {
-    Anchor['mysql::server::start'] ->
-    Class['mysql::server::install'] ->
-    Class['mysql::server::config'] ->
-    Class['mysql::server::service'] ->
-    Class['mysql::server::root_password'] ->
-    Class['mysql::server::providers'] ->
-    Anchor['mysql::server::end']
+    Class['mysql::server::service']
   }
 
+  Anchor['mysql::server::start'] ->
+  Class['mysql::server::install'] ->
+  Class['mysql::server::config'] ->
+  Class['mysql::server::installdb'] ->
+  Class['mysql::server::service'] ->
```

Comment:
do not look at the diff look at the plain file it is more clear. in the first case it is a ~> that happens only of $restart